var/spack/repos/builtin/packages/samrai/package.py

```diff
@@ -60,19 +60,26 @@ class Samrai(AutotoolsPackage):
     # and more print statements
     variant('debug', default=False,
             description='Compile with reduced optimization and debugging on')
+    variant('silo', default=False,
+            description='Compile with support for silo')
 
     depends_on('mpi')
     depends_on('zlib')
     depends_on('hdf5+mpi')
     depends_on('m4', type='build')
 
-    # Starting with 3.12.0, samrai no longer depends on boost.
-    # version 3.11.5 or earlier can only work with boost version
-    # 1.64.0 or earlier.
-    depends_on('boost@:1.64.0', when='@0:3.11.99', type='build')
+    depends_on('silo+mpi', when='+silo')
 
-    # don't build tools with gcc
-    patch('no-tool-build.patch', when='%gcc')
+    # Versions after 3.0 but and before (but not including) 3.12.0 depend on
+    # boost. Versions 3.0 - 3.11.5 can only work with boost version 1.64.0 or
+    # earlier.
```

Comment:
This comment is pretty useless. It just translates the line below to English. Not sure if it is necessary.

Reply:
Fixed.